test/hpe_docker_unit_test.py

```diff
@@ -94,6 +94,8 @@ def _mock_execute_api(self, mock_objects, plugin_api=''):
         req_body = self._get_request_body(self.get_request_params())
 
         _api = api.VolumePlugin(reactor, self._host_config, self._all_configs)
+        import time
+        time.sleep(1)
```

Comment:
In case where Plugin takes more time then one second, the unit test will end up with failure.

We can perhaps have a foolproof mechanism by checking initialization status of a backend.
get_request_params() return a dictionary which contains backend name as an optional parameter. If it is not there then we can assume "DEFAULT" backend.
In either case, can we check VolumePlugin for the initialization status of the intended backend?
```python
req_params = self.get_request_params()  
backend = req_params.get('backend', 'DEFAULT')  
while(True):  
    if _api.is_initialized(backend):  
        break  
    if _api.is_failed(backend):  
        # Return error  
   sleep(1)  
   continue
```

Reply:
@imran-ansari , i have done the backend state check in the new commit, please review. Thanks for your comment, now I have made change to list uninitialized backend (or in backend in process of initialization) as "INITIALIZING" state instead of earlier "FAILED" state